tools/fuse-waked/fuse-waked.cpp

```diff
@@ -1537,7 +1537,7 @@ int main(int argc, char *argv[]) {
    * This anti-feature was added in 2.4.0 and removed in 3.0.0.
    */
   if (fuse_opt_add_arg(&args, "wake") != 0 || fuse_opt_add_arg(&args, "-o") != 0 ||
-      fuse_opt_add_arg(&args, "nonempty") != 0) {
+      fuse_opt_add_arg(&args, "nonempty,auto_unmount") != 0) {
 #else
   if (fuse_opt_add_arg(&args, "wake") != 0) {
```

Comment:
Should we also add it to this code path? Otherwise things might regress when we eventually upgrade the FUSE version, since the `#if` block above only targets a very narrow range of versions.

Reply:
Whoops, I my GitHub hadn't loaded your comment when I first opened the diff.

Reply:
heh jynx again, I'm uncertain as my own comment on this states. I'd have to dig in the libfuse repo to see what the first version to support this is? I'm also down to just add it. I would imagine 2.9.7 is the oldest that we'll be dealing with.

Reply:
Not sure what the version number was, but it was first released in 2011, so I think we're probably fine: https://github.com/libfuse/libfuse/blob/d44bf3a4ac1ce1b56672ac9ae4a5caa3a8ee6844/ChangeLog#L223-L225

Although while digging through the changelog, I did find something that we should maybe investigate before merging this in: https://github.com/libfuse/libfuse/blob/d372d3f80b5437e652ea501d8a4a917f7001b9d7/ChangeLog.rst#libfuse-301-2017-04-10

> Document that -o auto_unmount implies -o nodev,nosuid.

I don't know off the top of my head if we rely on either of those things.

Reply:
```
Most of the generic mount options described in mount are
       supported (ro, rw, suid, nosuid, dev, nodev, exec, noexec, atime,
       noatime, sync, async, dirsync). Filesystems are mounted with
       nodev,nosuid by default, which can only be overridden by a
       privileged user.
   ```
   
Oh good catch but luckily it looks like that was already happening

Reply:
Oh, nice find. Yeah, this LGTM, then.